emcc

```diff
@@ -445,10 +445,16 @@ try:
 
   settings_changes = []
 
-  def validate_arg_level(level_string, max_level, err_msg):
+  def validate_arg_level(level_string, max_level, err_msg, clamp=False):
     try:
       level = int(level_string)
-      assert 0 <= level <= max_level
+      assert 0 <= level
+      if clamp:
+        if level > max_level:
+          logging.warning(err_msg + '; clamped to ' + str(max_level))
```

Comment:
perhaps we should not print `err_msg`? It seems like showing "Invalid optimization level" on a non-error might be confusing. we could add another optional parameter, "description" that we would print here, so something like "clamped [description] to"